src/components/app/index.tsx

```diff
@@ -25,33 +66,131 @@ export default class App extends Component<Props, State> {
     }
   }
 
+  getEncoderName(type: ImageType, options: CodecOptions) {
+    const name = EncoderNames[type];
+    if (typeof name === 'function') return name(options);
+    return name;
+  }
+
+  @bind
+  setLeftType(leftType: ImageType) {
+    this.setState({ leftType, leftOptions: {} });
+    this.optionsUpdated();
+  }
+
+  @bind
+  setRightType(rightType: ImageType) {
+    this.setState({ rightType, rightOptions: {} });
+    this.optionsUpdated();
+  }
+
+  @bind
+  setLeftOptions(leftOptions: any) {
+    this.setState({ leftOptions });
+    this.optionsUpdated();
+  }
+
+  @bind
+  setRightOptions(rightOptions: any) {
+    this.setState({ rightOptions });
+    this.optionsUpdated();
+  }
+
+  optionsUpdated() {
+    if (!this.optionsUpdateTimer) {
+      this.optionsUpdateTimer = setTimeout(this.commitOptionsUpdated, 500);
+    }
+  }
```

Comment:
I'd like us to have a think about the best kind of debouncing to use in these cases, but not needed for this PR.